content/atomics_intro.tex

```diff
@@ -4,68 +4,102 @@
+To force completion for these non-fetching atomic routines,
+\FUNC{shmem\_quiet}, \FUNC{shmem\_barrier}, or \FUNC{shmem\_barrier\_all} can
+be used by an \openshmem{} program.
 
-The \textit{non-fetching} operations include: \FUNC{SHMEM\_SET}, \FUNC{SHMEM\_INC} and
-\FUNC{SHMEM\_ADD}.
+The non-fetching operations include:
+\FUNC{shmem\_atomic\_\{set, inc, add, and, or, xor\}}.
 \end{itemize}
 
-Where appropriate compiler support is available, \openshmem{} provides type-generic
-atomic memory operation interfaces via \Celev{} generic selection. The type-generic 
-\ac{AMO} routines each support the ``standard \ac{AMO} types’’ listed in Table \ref{stdamotypes}, 
-except for \FUNC{shmem\_fetch}, \FUNC{shmem\_set}, and \FUNC{shmem\_swap}, which supports the ``extended \ac{AMO} types’’ listed 
-in Table \ref{extamotypes}.
+Where appropriate compiler support is available, \openshmem{} provides
+type-generic \ac{AMO} interfaces via \Celev{} generic selection.
+The type-generic support for the \ac{AMO} routines is as follows:
+
+\begin{itemize}
+\item \FUNC{shmem\_atomic\_\{compare\_swap, fetch\_inc, inc\}} support
+      the ``standard \ac{AMO} types'' listed in Table~\ref{stdamotypes},
+\item \FUNC{shmem\_atomic\_\{fetch, set, swap, fetch\_add, add\}} support
+      the ``extended \ac{AMO} types'' listed in Table~\ref{extamotypes}, and
+\item \FUNC{shmem\_atomic\_\{fetch\_and, and, fetch\_or, or, fetch\_xor, xor\}}
+      support the ``bitwise \ac{AMO} types'' listed in Table~\ref{bitamotypes}.
+\end{itemize}
 
 \begin{table}[h]
   \begin{center}
     \begin{tabular}{|l|l|}
       \hline
-      \TYPE & \TYPENAME\\
-      \hline
-      int & int\\
-      \hline
-      long & long\\
-      \hline
-      long long & longlong\\
-      \hline
+      \TYPE              & \TYPENAME  \\ \hline
+      int                & int        \\ \hline
+      long               & long       \\ \hline
+      long long          & longlong   \\ \hline
+      unsigned int       & uint       \\ \hline
+      unsigned long      & ulong      \\ \hline
+      unsigned long long & ulonglong  \\ \hline
+      int32\_t           & int32      \\ \hline
+      int64\_t           & int64      \\ \hline
+      uint32\_t          & uint32     \\ \hline
+      uint64\_t          & uint64     \\ \hline
+      size\_t            & size       \\ \hline
+      ptrdiff\_t         & ptrdiff    \\ \hline
     \end{tabular}
     \caption{Standard \ac{AMO} Types and Names}
     \label{stdamotypes}
-  \end{center} 
+  \end{center}
 \end{table}
 
 \begin{table}[h]
   \begin{center}
     \begin{tabular}{|l|l|}
       \hline
-      \TYPE & \TYPENAME\\
-      \hline
-      float & float\\
-      \hline
-      double & double\\
-      \hline
-      int & int\\
-      \hline
-      long & long\\
-      \hline
-      long long & longlong\\
-      \hline
+      \TYPE              & \TYPENAME  \\ \hline
+      float              & float      \\ \hline
+      double             & double     \\ \hline
```

Comment:
It's just one data point, but -- as you noted in conversation and I verified -- the Intel x86 ISA shows that the LOCK prefix may not be used on float and double types. I'll remove floating-point AMO support.